### PR TITLE
Enforce TypeScript checks and fix types

### PR DESCRIPTION
**Pull request by mareklibra**

### CODE REVIEW

**Review by rawagner — APPROVED** [review]

**Review by jtomasek — DISMISSED** [review]

File: src/common/components/i18n/i18n.tsx [hunk 1]
```diff
@@ -1,4 +1,6 @@
-/*
+export const DO_NOTHING = '';
+
+/* We keep that for future reuse. Internationalization has been postponed.
```
Comment (jkilzi) [inline src/common/components/i18n/i18n.tsx:3]: Why are we still keeping this file???

File: src/cim/components/ClusterDeployment/ClusterDeploymentDetailsStep.tsx [hunk 2]
```diff
@@ -110,7 +118,7 @@ const ClusterDeploymentDetailsStep: React.FC<ClusterDeploymentDetailsStepProps>
     } catch (error) {
       addAlert({
         title: 'Failed to save ClusterDeployment',
-        message: error,
+        message: error as string,
```
Comment (jkilzi) [inline src/cim/components/ClusterDeployment/ClusterDeploymentDetailsStep.tsx:121]: How do you know `error` is a string and not an object?

**Review by jkilzi — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -38,7 +38,7 @@
     "start-uhc-portal": "./scripts/start-uhc-portal.sh",
     "sync-to-ui": "./scripts/sync-dist.sh",
     "prepare": "install-peers",
-    "test": "run-s test:circular test:lint",
+    "test": "run-s test:circular test:lint test:type-check",
```
[the remark this answers is not in the document]
Comment (mareklibra) [reply package.json:41]: We can but typechecking should be mandatory part of the developer flow even locally, that will enforce it.

Secondly, that target is kind of workaround, it should be handled by esbuild's plugin which unfortunately does not work as expected.

File: .gitignore [hunk 4]
```diff
@@ -18,7 +18,7 @@ docs
 .env
 .env.local
 .env.*.local
-.vscode/*
+.vscode
```
[the remark this answers is not in the document]
Comment (mareklibra) [reply .gitignore:21]: It's just IDE configuration, unrelated to the project.
I am ok with keeping it if everybody agrees but those config files are usually removed from the source versioning in other projects.

And I see them changed quite regularly without my explicit will.

[the remark this answers is not in the document]
Comment (mareklibra) [reply .gitignore:21]: I will remove that change from this PR and push it separately. Let's discuss there ...

File: .vscode/tasks.json [hunk 5]
```diff
@@ -1,58 +0,0 @@
-{
```
[the remark this answers is not in the document]
Comment (mareklibra) [reply .vscode/tasks.json:1]: see bellow

File: src/common/components/i18n/i18n.tsx (hunk 1 above)
Comment (mareklibra) [reply src/common/components/i18n/i18n.tsx:3]: Right, we can remove it.
It was commented-out, proposed change is to keep existing state and making it just compilable.

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (jkilzi) [reply package.json:41]: Yep, unfortunately [esbuild doesn't](https://news.ycombinator.com/item?id=24097984) [perform typechecking](https://esbuild.github.io/content-types/#typescript-caveats) and [the plug-in](https://www.npmjs.com/package/esbuild-plugin-d.ts) only generates type declarations (.d.ts files), it doesn't perform type-checking neither... but [swc](https://swc.rs/) does :)

**Review by jkilzi — APPROVED** [review]

**Pull request merged**